components/constellation/constellation.rs

```diff
@@ -1862,17 +1892,20 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
         // and pass the event to that script thread.
         // If the pipeline lookup fails, it is because we have torn down the pipeline,
         // so it is reasonable to silently ignore the event.
-        let browsing_context_id = self.pipelines.get(&pipeline_id).map(|pipeline| pipeline.browsing_context_id);
+        let top_level_browsing_context_id = self.pipelines.get(&pipeline_id)
```

Comment:
Yes, this is one of those odd cases that the extra typechecking helped with. Yay types!